static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts

```diff
@@ -77,69 +78,62 @@ const setPublisherProvidedId = (): void =>
-			},
-		);
+		return getEnhancedConsent().then(
+			(consentState: ConsentStateEnhanced) => {
+				let canRun = true;
 
-		return getInitialConsentState().then((state) => {
-			let canRun = true;
-			if (state.ccpa) {
-				const doNotSell = state.ccpa.doNotSell;
-				// CCPA mode
-				window.googletag.cmd.push(() => {
-					window.googletag.pubads().setPrivacySettings({
-						restrictDataProcessing: doNotSell,
-					});
-				});
-				if (!state.ccpa.doNotSell) {
+				if (consentState.canTarget) {
 					window.googletag.cmd.push(setPublisherProvidedId);
 				}
-			} else {
-				if (state.tcfv2) {
-					// TCFv2 mode
-					const canTarget = Object.values(state.tcfv2.consents).every(
-						Boolean,
-					);
-					if (canTarget) {
-						window.googletag.cmd.push(setPublisherProvidedId);
-					}
 
-					canRun = getConsentFor('googletag', state);
-				} else if (state.aus) {
+				if (consentState.ccpa) {
+					// CCPA mode
+					// canRun stays true, set RDP flag
+					window.googletag.cmd.push(() => {
+						window.googletag.pubads().setPrivacySettings({
+							restrictDataProcessing: !consentState.canTarget,
+						});
+					});
+				} else if (consentState.tcfv2) {
+					// TCFv2 mode
+					canRun = getConsentFor('googletag', consentState);
+				} else if (consentState.aus) {
 					// AUS mode
-					// canRun stays true, set NPA flag if consent is retracted
-					const npaFlag = !getConsentFor('googletag', state);
+					// canRun stays true, set NPA flag
+					const npaFlag = !getConsentFor('googletag', consentState);
 					window.googletag.cmd.push(() => {
 						window.googletag
 							.pubads()
 							.setRequestNonPersonalizedAds(npaFlag ? 1 : 0);
 					});
-					if (!npaFlag) {
-						window.googletag.cmd.push(setPublisherProvidedId);
-					}
 				}
-			}
-			// Prebid will already be loaded, and window.googletag is stubbed in `commercial.js`.
-			// Just load googletag. Prebid will already be loaded, and googletag is already added to the window by Prebid.
-			if (canRun) {
-				void loadScript(
-					config.get<string>(
-						'libs.googletag',
-						'//www.googletagservices.com/tag/js/gpt.js',
-					),
-					{ async: false },
-				);
-			}
-			return Promise.resolve();
-		});
+
+				// Prebid will already be loaded, and window.googletag is stubbed in `commercial.js`.
+				// Just load googletag. Prebid will already be loaded, and googletag is already added to the window by Prebid.
+				if (canRun) {
+					// note: fillAdvertSlots isn't synchronous like most buffered cmds, it's a promise. It's put in here to ensure
+					// it strictly follows preceding prepare-googletag work (and the module itself ensures dependencies are
+					// fulfilled), but don't assume fillAdvertSlots is complete when queueing subsequent work using cmd.push
+					window.googletag.cmd.push(
+						setDfpListeners,
+						setPageTargeting(consentState),
+						refreshOnResize,
+						() => {
+							void fillAdvertSlots();
+						},
+					);
```

Comment:
The setup of `window.googletag.cmd` has been moved inside the consent state promise.

afaics everything that makes use of page targeting on the window has to first get consent state anyway - i.e. prebid, appNexus & YouTube.

Reply:
I moved the setup of `window.googletag.cmd` back outside of `getInitialConsentState` as the appnexus script stopped working.

For now this PR fixes the initial bug and can be released as is. 

Fixing the appnexus script initialisation and making consent state a mandatory parameter to `getPageTargeting` can be tackled in another PR.